### ISSUE TITLE
Wrong reporting with nested suites with same name

### ISSUE THREAD

**Issue by maartenth** [webdriverio/webdriverio#1742]
## The problem

If you have nested suites and an innermost suite (A) has the same name as the innermost suite (B) in another parent suite, the test results are erroneously overwritten and not presented in the right order in all reporters.

## Environment

* WebdriverIO version: 4.4.0
* Node.js version: doesn't matter
* [Standalone mode or wdio testrunner](http://webdriver.io/guide/getstarted/modes.html): both
* if wdio testrunner, running synchronous or asynchronous tests: sync
* Additional wdio packages used (if applicable): no

## Details

Minimal example:

```
describe('Scenario 1', function () {
    describe('Part 1', function () {
        it('should work (inside scenario 1)', function () {
            browser.url('/');
        });
    } );
});

describe('Scenario 2', function () {
    describe('Part 1', function () {
        it('should work (inside scenario 2)', function () {
            browser.url('/');
        });
    } );
});
```

Output:

![screen shot 2016-12-01 at 12 06 58](https://cloud.githubusercontent.com/assets/298893/20791842/3b29ca44-b7bf-11e6-8a0d-265163d5f0d6.png)

Note that the result of the _second_ `it` is posted under the _first_ `describe`, and that the first `it` is nowhere to be found.

The reason for this bug is that in `ReporterStats.js`, one object is used for all test suites. There are 2 possible solutions:
- use an array instead of an object
- use an object with nested subsuite properties that properly represents the suite nesting

**Comment by christian-bromann** [comment]
@maartenth or do
```js
describe('test', function () {
    describe('Scenario 1', function () {
        describe('Part 1', function () {
            it('should work (inside scenario 1)', function () {
                browser.url('/');
            });
        });
    });

    describe('Scenario 2', function () {
        describe('Part 1', function () {
            it('should work (inside scenario 2)', function () {
                browser.url('/');
            });
        });
    });
});
```

**Comment by christian-bromann** [comment]
This will be fixed with #1717. Closing.
If you have further questions please join our  [![Gitter](https://badges.gitter.im/webdriverio/webdriverio.svg)](https://gitter.im/webdriverio/webdriverio?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) or refer to #1717

**Comment by maartenth** [comment]
Thank you!
